maps_dashboards/public/components/map_container/map_container.tsx

```diff
@@ -95,6 +102,34 @@ export const MapContainer = ({
     };
   }, [layers]);
 
+  // Handle map bounding box change, it should update the search if "request data around map extent" was enabled
+  useEffect(() => {
+    function renderLayers() {
+      layers.forEach((layer: MapLayerSpecification) => {
+        // We don't send search query if the layer doesn't have "request data around map extent" enabled
+        if (
+          layer.type === DASHBOARDS_MAPS_LAYER_TYPE.OPENSEARCH_MAP ||
+          !layer.source.useGeoBoundingBoxFilter
+        ) {
+          return;
+        }
+        handleDataLayerRender(layer, mapState, services, maplibreRef, undefined);
+      });
+    }
```

Comment:
nit: Since we are adding Custom Map in another PR.
```suggestion
  // Handle map bounding box change, it should update the search if "request data around map extent" was enabled
  useEffect(() => {
    function renderLayers() {
      layers.forEach((layer: MapLayerSpecification) => {
        // We don't send search query if the layer doesn't have "request data around map extent" enabled
        if (
          layer.type === DASHBOARDS_MAPS_LAYER_TYPE.DOCUMENTS &&
          layer.source.useGeoBoundingBoxFilter
        ) {
          handleDataLayerRender(layer, mapState, services, maplibreRef, undefined);
        }
      });
    }
```

Reply:
@VijayanB Would there be other type of data layers other than "Documents" which need to have bounding box query enabled?

Reply:
Good question. Yes, in subsequent release, not at this moment. I am thinking about creating common type for all layers and create flag to identify whether they support feature or not, to avoid type checking everywherere.